enip_backend/export/national.py

```diff
@@ -0,0 +1,422 @@
+            if data.dem:
+                # There is already a Dem candidate. We process in order of number
+                # of votes, so this is a secondary Dem candidate
+                if hasattr(data, "multiple_dem"):
+                    data.multiple_dem = True
+            else:
+                # This is the leading Dem candidate
+                data.dem = named_candidate_factory(
+                    first_name=record.first,
+                    last_name=record.last,
+                    pop_vote=record.votecount,
+                    pop_pct=record.votepct,
+                    pop_vote_history=self.historical_counts.get(record.elex_id, {}),
+                )
+                return
+
+        # Third-party candidate or non-leading gop/dem
+        data.oth.pop_vote += record.votecount
+        data.oth.pop_pct += record.votepct
+
+        # Merge the candidate's historical counts into the overall historical
+        # counts
+        for datetime_str, count in self.historical_counts.get(
+            record.elex_id, {}
+        ).items():
+            if datetime_str in data.oth.pop_vote_history:
+                data.oth.pop_vote_history[datetime_str] += count
+            else:
+                data.oth.pop_vote_history[datetime_str] = count
+
+    def grant_electoral_votes(self, party: structs.Party, count: int):
+        """
+        Helper function to add electoral votes to the national summary when
+        we've called a state for a candidate
+        """
+        race = self.data.national_summary.P
+        if party == structs.Party.DEM:
+            assert race.dem is not None
+            race.dem.elect_won += count
+        elif party == structs.Party.GOP:
+            assert race.gop is not None
+            race.gop.elect_won += count
+        else:
+            race.oth.elect_won += count
+
+    def grant_congressional_seat(
+        self, data: structs.NationalSummaryWinnerCount, party: structs.Party
+    ):
+        """
+        Helper function to add one to the senate/house national summary
+        """
+        if party == structs.Party.DEM:
+            data.dem.won += 1
+        elif party == structs.Party.GOP:
+            data.gop.won += 1
+        else:
+            data.oth.won += 1
+
+    def record_ntl_result(self, record: SQLRecord) -> None:
+        """
+        Records a "national"-level result. These results are just the national-level
+        results for a presidential candidate
+        """
+        assert record.officeid == "P"
+        assert record.statepostal == "US"
+
+        self.handle_candidate_results(
+            self.data.national_summary.P,
+            structs.NationalSummaryPresidentCandidateNamed,
+            record,
+        )
+
+    def record_district_result(self, record: SQLRecord) -> None:
+        """
+        Records a "district"-level result. These results are presidential results
+        for congression district and at-large NE and ME districts.
```

Comment:
somehow, "congression" isn't in the Scrabble dictionary! weird I wonder how "congressional" became an idiomatic word.